src/server/modes/charts/plugins/datalens/preparers/line/highcharts.ts

```diff
@@ -0,0 +1,192 @@
+import _isEmpty from 'lodash/isEmpty';
+
+import {
+    ChartkitHandlers,
+    MINIMUM_FRACTION_DIGITS,
+    PlaceholderId,
+    ServerField,
+    WizardVisualizationId,
+    getAxisMode,
+    getFakeTitleOrTitle,
+    getIsNavigatorEnabled,
+    isDateField,
+    isFloatField,
+    isMeasureNameOrValue,
+    isVisualizationWithSeveralFieldsXPlaceholder,
+} from '../../../../../../../shared';
+import {getFieldExportingOptions} from '../../utils/export-helpers';
+import {getAxisType} from '../helpers/axis';
+import {
+    getHighchartsColorAxis,
+    isXAxisReversed,
+    shouldUseGradientLegend,
+} from '../helpers/highcharts';
+import {getYPlaceholders} from '../helpers/layers';
+import {getSegmentMap} from '../helpers/segments';
+import {PrepareFunctionArgs} from '../types';
+
+import {getSegmentsYAxis} from './helpers';
+import {getAxisFormattingByField} from './helpers/axis/getAxisFormattingByField';
+import {prepareLineData} from './prepare-line-data';
+
+// eslint-disable-next-line complexity
+function getHighchartsConfig(args: PrepareFunctionArgs & {graphs: any[]}) {
+    const {placeholders, colors, colorsConfig, sort, visualizationId, shared, shapes, graphs} =
+        args;
+    const xPlaceholder = placeholders.find((p) => p.id === PlaceholderId.X);
+    const xPlaceholderSettings = xPlaceholder?.settings;
+    const x: ServerField | undefined = xPlaceholder?.items[0];
+    const isXDiscrete = getAxisMode(xPlaceholderSettings, x?.guid) === 'discrete';
```

Comment:
Probably now we should use enum value for 'discrete'
We have one constant in src/ui/constants, called DATE_MODES
I think we can rename it or use more common enum